### ISSUE TITLE
resource/aws_iam_user_ssh_key: Add Computed attributes for both encodings, adjust encoding attribute documentation for its purpose

### ISSUE THREAD

**Issue by alewando** [hashicorp/terraform-provider-aws#7036]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.11.11
+ provider.aws v1.54.0
+ provider.tls v1.2.0

### Affected Resource(s)

* aws_iam_user_ssh_key

### Terraform Configuration Files

```hcl
provider "aws" {
  region = "us-east-1"
}

resource "aws_iam_user" "user" {
  name = "test-ssh-key"
}

resource "tls_private_key" "rsa-key" {
  algorithm = "RSA"
  rsa_bits = 4096
}

resource "aws_iam_user_ssh_key" "ssh-key" {
  username   = "${aws_iam_user.user.name}"
  encoding   = "SSH"
  public_key = "${tls_private_key.rsa-key.public_key_pem}"
}
```

### Debug Output

https://gist.github.com/alewando/5ba065f80ad64b65626c8c72256b6cbd

### Expected Behavior

Executing `terraform apply` repeatedly should be idempotent if there are no changes. No resources should be dropped or created on executions subsequent to the first.

### Actual Behavior
`terraform apply` or `terraform plan` shows that the public key is forcing a new resource and has to be dropped/recreated. 

### Steps to Reproduce

1. `terraform apply` to create resources (user, ssh key)
2. `terraform plan` to verify resources are in correct state, and no changes are needed.
3. Note that plan shows that a new ssh key is needed, even though nothing has changed

### References

I believe the relevant commit which triggered this behavior is 6bafd806aed240a0a43ca6ae9a745aa65f36f5fd

**Comment by bflad** [comment]
Hi @alewando 👋 Sorry for the trouble here. Previously the `aws_iam_user_ssh_key` resource did not perform drift detection with the contents of the `public_key` attribute until #6727 released in version 1.52.0 of the Terraform AWS provider. The current drift detection relies on correctly setting the `encoding` attribute to match the encoding of the `public_key` attribute.

Since the value for the `public_key` attribute you are submitting is in PEM encoding from the `tls_private_key.rsa-key.public_key_pem` reference, you should be able to change your `aws_iam_user_ssh_key` resource configuration to use `encoding = "PEM"` to "fix" the Terraform difference and prevent recreation:

```hcl
resource "aws_iam_user_ssh_key" "ssh-key" {
  username   = "${aws_iam_user.user.name}"
  encoding   = "PEM"
  public_key = "${tls_private_key.rsa-key.public_key_pem}"
}
```

However, if you need the SSH encoded public key contents (e.g. referencing `aws_iam_user_ssh_key.ssh-key.public_key` later with the intent to grab the SSH encoded version), you'll likely need to use an `ignore_changes` workaround:

```hcl
resource "aws_iam_user_ssh_key" "ssh-key" {
  username   = "${aws_iam_user.user.name}"
  encoding   = "SSH"
  public_key = "${tls_private_key.rsa-key.public_key_pem}"

  lifecycle {
    ignore_changes = ["public_key"]
  }
}
```

In the second scenario, we could add an enhancement to the resource to always return both encodings in new, separate attributes (e.g. `pem_encoded_key` and `ssh_encoded_key`) so they are available for downstream references regardless of how what encoding was used for uploading (e.g. still using `encoding` for drift detection with `public_key`).

Can you let us know if either of these help your case? Thanks.

**Comment by alewando** [comment]
Thank you for the explanation Brian. Changing the encoding attribute on the ssh key resource to match  the source encoding (PEM) did the trick. We do not reference this ssh_key resource anywhere else; all other references were directly to the `tls_private_key.public_key_pem` attribute itself. 

I do agree that adding separate outputs for the different encoding formats (`pem_encoded_key`, `openssh_encoded_key`) would be helpful. Using an argument to control the format of an attribute is less clear (to me, at least) than having separate output attributes (similar to what `tls_private_key` does itself).
